tests/unit/components/playback/controls/EditCustomer.spec.js

```diff
@@ -0,0 +1,514 @@
+      "06ad547f fe02-477b-9473-f7977e4d5e17",
+      "missing hypen",
+      "apikey-id",
+      "validate_alphanumeric",
+    ],
+    [
+      "06ad547f-fe02-477b-9473-f7977e4d5e1",
+      "less than 36",
+      "apikey-id",
+      "validate_alphanumeric",
+    ],
+    [
+      "06ad547f-fe02-477b-9473-f7977e4d5e14k",
+      "more than 36",
+      "apikey-id",
+      "validate_alphanumeric",
+    ],
+    ["Experiment anemia -1", "valid input", "nickname-id", "validate_nickname"],
+    ["Cat * lab", "contains asterisk *", "nickname-id", "validate_nickname"],
+    ["Cat lab`", "contains left quote `", "nickname-id", "validate_nickname"],
+    ["Cat lab;", "contains semi-colon ;", "nickname-id", "validate_nickname"],
+    [
+      "Experiment anemia alpha cells -1",
+      "more than 20 characters",
+      "nickname-id",
+      "validate_nickname",
+    ],
+    ["C", "minimum one character C", "nickname-id", "validate_nickname", ""],
+  ])(
+    "When the text %s (%s) is entered into the field found with the selector ID %s, Then the correct text validation function (%s) is called and the error message from the validation function is rendered below the input in the DOM",
+    async (
+      entry,
+      test_description,
+      selector_id_suffix,
+      text_validation_type
+    ) => {
+      const editcustomer = {
+        uuid: "",
+        apikey: "",
+        nickname: "",
+      };
+
+      if (text_validation_type === "validate_uuidBase_fiftyseven_encode") {
+        editcustomer.uuid = entry;
+      }
+      if (text_validation_type === "validate_alphanumeric") {
+        editcustomer.apikey = entry;
+      }
+      if (text_validation_type === "validate_nickname") {
+        editcustomer.nickname = entry;
+      }
+
+      const propsData = {
+        dialogdata: editcustomer,
+        dataindex: 0,
+      };
+      wrapper = mount(ComponentToTest, {
+        propsData,
+        store,
+        localVue,
+      });
+
+      const spied_text_validator = jest.spyOn(
+        TextValidation.prototype,
+        text_validation_type
+      );
+
+      const target_input_field = wrapper.find(
+        "#input-widget-field-" + selector_id_suffix
+      );
+
+      const target_error_message = wrapper.find(
+        "#input-widget-feedback-" + selector_id_suffix
+      );
+
+      target_input_field.setValue(entry);
+
+      await Vue.nextTick();
+      expect(spied_text_validator).toHaveBeenCalledWith(entry);
+
+      expect(target_error_message.text()).toStrictEqual(
+        spied_text_validator.mock.results[0].value
+      );
+    }
+  );
+});
+
+describe("EditCustomer.enable_save_button", () => {
+  beforeEach(async () => {
+    store = await NuxtStore.createStore();
+  });
+
+  beforeAll(async () => {
+    // note the store will mutate across tests, so make sure to re-create it in beforeEach
+    const storePath = `${process.env.buildDir}/store.js`;
+    NuxtStore = await import(storePath);
+  });
+
+  afterEach(() => wrapper.destroy());
```

Comment:
this call to destroy the wrapper is common to every single test in this spec file correct? If so, then it should be in a `describe` that wraps all the tests so that it is not duplicated.  Check if this comment is applicable to the other 3 spec files also

Reply:
Corrected to contain per `describe` testcase in all the spec.js files.